src/main/java/seedu/address/logic/commands/GradeProgressCommand.java

```diff
@@ -0,0 +1,101 @@
+package seedu.address.logic.commands;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.GradeProgress;
+import seedu.address.model.person.GradeProgressList;
+import seedu.address.model.person.Person;
+
+/**
+ * Adds homework to an existing person in the address book.
+ */
+public class GradeProgressCommand extends Command {
+
+    public static final String COMMAND_WORD = "gradeProgress";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Adds grade progress to the person identified "
+            + "by the index number used in the last person listing. "
+            + "Existing grade progress will not be modified.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "gp/ [GRADEPROGRESS]\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "gp/ MATH:A";
```

Comment:
maybe you can use the const prefix here just for consistency's sake with the other command classes

Reply:
Noted, thank you